db/music_pg.sql

```diff
@@ -0,0 +1,373 @@
+  "participant_id" int NOT NULL,
+  "nickname" text NOT NULL,
+  "date_start" date NOT NULL,
+  "date_end" date
+);
+
+CREATE TABLE "genre" (
+  "genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text UNIQUE NOT NULL,
+  "parent_id" text
+);
+
+CREATE TABLE "song_genre" (
+  "song_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "song_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "bracket" (
+  "bracket_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "epoch" int,
+  "cycle" int,
+  "name" text,
+  "type" text NOT NULL,
+  "year_min" int,
+  "year_max" int,
+  "order" int NOT NULL,
+  "size" text NOT NULL,
+  "date_start" timestamp,
+  "date_end" timestamp,
+  "discord_message_id" text
+);
+
+CREATE TABLE "project" (
+  "project_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text UNIQUE,
+  "project_lead_id" int
+);
+
+CREATE TABLE "bracket_selection" (
+  "bracket_selection_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "project_id" int NOT NULL,
+  "epoch" int NOT NULL,
+  "cycle" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "date_selected" timestamp NOT NULL
+);
+
+CREATE TABLE "seed" (
+  "seed_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "bracket_id" int NOT NULL,
+  "seed" int NOT NULL,
+  "submission_id" int NOT NULL,
+  "emoji" text
+);
+
+CREATE TABLE "seed_history" (
+  "seed_history_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "date_changed" timestamp NOT NULL,
+  "reason_changed" text NOT NULL,
+  "seed_id" int NOT NULL,
+  "submission_id" int NOT NULL,
+  "emoji" text
+);
+
+CREATE TABLE "round" (
+  "round_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "bracket_id" int NOT NULL,
+  "name" text NOT NULL,
+  "size" int NOT NULL,
+  "index_asc" int NOT NULL,
+  "index_desc" int NOT NULL,
+  "date_start" timestamp NOT NULL,
+  "date_end" timestamp,
+  "discord_message_id" text
+);
+
+CREATE TABLE "match" (
+  "match_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "round_id" int NOT NULL,
+  "index" int NOT NULL,
+  "date_posted" timestamp NOT NULL,
+  "tie" text,
+  "winner_id" int,
+  "date_cancelled" timestamp,
+  "reason_cancelled" text,
+  "next_match_id" int,
+  "discord_message_id" text
+);
+
+CREATE TABLE "match_seed" (
+  "match_seed_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "match_id" int NOT NULL,
+  "seed_id" int NOT NULL,
+  "index" int NOT NULL,
+  "votes" int,
+  "voters" text[],
+  "previous_match_id" int
+);
+
+CREATE TABLE "notifications" (
+  "notification_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "submission_id" int,
+  "date_first_attempt" timestamp NOT NULL,
+  "date_succeded" timestamp,
+  "type" text NOT NULL,
+  "discord_message_id" text
+);
+
+CREATE TABLE "config" (
+  "project_id" int NOT NULL,
+  "key" text,
+  "value" text,
+  PRIMARY KEY ("project_id", "key")
+);
+
+CREATE TABLE "seed_order" (
```

Comment:
I think we'll want to do away with this entirely, we can have these orders hardcoded in some script that imports sheet data into the DB, but I'd like to have this project have no real notion of that additional translation. The `seed` table should be able to handle whatever we need for ordering songs into matches and whatnot here. When we construct a seeding for a given bracket, we'd just make new `seed` rows for each submission (beyond the size of the bracket if we like to contain the spillover and have replacements lined up) and it should Just Work.

Reply:
That's reasonable.